[nnc] Refactored test macros and updated compress buffer tests to use them

navahgar opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack):
* **#61716**
* #61715

Differential Revision: [D29715754](https://our.internmc.facebook.com/intern/diff/D29715754)

navahgar committed:
[nnc] Refactored test macros and updated compress buffer tests to use them

[ghstack-poisoned]

navahgar commented:
@navahgar has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D29715754).

ZolotukhinM reviewed (approved):
Nice, thanks for the follow-up!

navahgar committed:
Update on "[nnc] Refactored test macros and updated compress buffer tests to use them"


Differential Revision: [D29715754](https://our.internmc.facebook.com/intern/diff/D29715754)

[ghstack-poisoned]

navahgar commented:
@navahgar has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D29715754).
